ckanext/xloader/loader.py

```diff
@@ -118,7 +118,10 @@ def _clear_datastore_resource(resource_id):
     engine = get_write_engine()
     connection = engine.connect()
     try:
-        connection.execute('TRUNCATE TABLE "{}"'.format(resource_id))
+        connection.execute("""
+            SET LOCAL lock_timeout = '5s';
+            TRUNCATE TABLE "{}"
```

Comment:
What about drop also?

Reply:
That's trickier since it's using a core feature to drop tables, not assembling its own query. We would have to alter our CKAN fork, not just the XLoader.

However, the majority of cases should be able to use the TRUNCATE logic. Changing the structure of data is much less common than merely uploading new data - especially on heavily trafficked resources, where it really matters.